pkg/api/server/audit/store/s3.go

```diff
@@ -0,0 +1,77 @@
+package store
+
+import (
+	"bytes"
+	"compress/gzip"
+	"context"
+	"errors"
+	"fmt"
+	"io"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/service/s3"
+)
+
+type S3StoreOptions struct {
+	AuditLogsStoreS3Bucket   string `usage:"Audit log store S3 bucket"`
+	AuditLogsStoreS3Endpoint string `usage:"Audit log store S3 endpoint"`
+}
+
+type s3Store struct {
+	host, bucket string
+	compress     bool
+	client       *s3.Client
+}
+
+func NewS3Store(host string, compress bool, options S3StoreOptions) (Store, error) {
+	if options.AuditLogsStoreS3Bucket == "" {
+		return nil, errors.New("Audit log store S3 bucket is required")
+	}
+
+	cfg, err := config.LoadDefaultConfig(context.Background())
+	if err != nil {
+		return nil, err
+	}
+
+	client := s3.NewFromConfig(cfg, func(o *s3.Options) {
```

Comment:
I assume the requirement is to have aws env variable setup?

Reply:
Yeah, the standard `AWS_` environment variables are required.